engine/start.php

```diff
@@ -24,6 +24,15 @@
 global $START_MICROTIME;
 $START_MICROTIME = microtime(true);
 
+// we need to register for shutdown before Symfony registers the session_write_close() function
+// https://github.com/Elgg/Elgg/issues/9243
+register_shutdown_function(function () {
+	// there are cases where this might not be defined so check first
```

Comment:
Later in start.php we have a couple opportunities to exit(). In 2.0 you can create an Application but never load the core (I think).